exporters/trace/jaeger/agent.go

```diff
@@ -124,6 +132,46 @@ func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error
 	return err
 }
 
+// calcSizeOfSerializedThrift calculate the serialized thrift packet size
+func (a *agentClientUDP) calcSizeOfSerializedThrift(ctx context.Context, thriftStruct thrift.TStruct) int {
+	a.thriftBuffer.Reset()
+	_ = thriftStruct.Write(ctx, a.thriftProtocol)
+	return a.thriftBuffer.Len()
+}
+
+// splitAndFlush split large batch into small ones to fit the udp packet limit
+func (a *agentClientUDP) splitAndFlush(ctx context.Context, batch *gen.Batch) error {
+	var err error
+	processSize := a.calcSizeOfSerializedThrift(ctx, batch.Process)
+	totalSize := processSize
+	var spans []*gen.Span
+	for _, span := range batch.Spans {
+		spanSize := a.calcSizeOfSerializedThrift(ctx, span)
+		if spanSize+processSize >= a.maxPacketSize {
+			// drop the span that exceeds the limit
+			continue
+		}
+		if totalSize+spanSize >= a.maxPacketSize {
+			err = a.Flush(ctx, &gen.Batch{
+				Process: batch.Process,
+				Spans:   spans,
+			})
+			spans = spans[:0]
+			totalSize = processSize
+		}
+		totalSize += spanSize
+		spans = append(spans, span)
+	}
+
+	if len(spans) > 0 {
+		err = a.Flush(ctx, &gen.Batch{
```

Comment:
This and the other place `err` is assigned to have the possibility to hide errors by setting the value back to `nil`.

We should return all errors encountered to provide the user with more information of the failure so things do not have to be serially debugged. We should likely store all the errors in slice and handle that slice appropriately when we return.

Reply:
> This and the other place `err` is assigned to have the possibility to hide errors by setting the value back to `nil`.
> 
> We should return all errors encountered to provide the user with more information of the failure so things do not have to be serially debugged. We should likely store all the errors in slice and handle that slice appropriately when we return.

It's ok to return a slice of errors, but the `EmitBatch` only one return vale of error. Returning a new error like "partial failure"?

Reply:
> It's ok to return a slice of errors, but the `EmitBatch` only one return vale of error. Returning a new error like "partial failure"?

Yeah, I would keep track of all errors seen in a slice and if there are more than one try and combine them into a "multiple errors during transform" sort of error.

Reply:
@MrAlias I've made some changes to handle your suggestions. Please have a look again